DB/src/main/java/io/deephaven/stream/StreamToTableAdapter.java

```diff
@@ -46,11 +143,65 @@ public void close() {
     public void refresh() {
         streamPublisher.flush();
         // Switch columns, update index, deliver notification
+
+        final long oldSize = index.size();
+        final long newSize;
+
+        synchronized (this) {
+            newSize = bufferChunkSources == null ? 0 : bufferChunkSources[0].getSize();
+
+            if (oldSize == 0 && newSize == 0) {
+                return;
+            }
+
+            ChunkColumnSource<?> [] oldPrev = prevChunkSources;
+
+            for (int ii = 0; ii < switchSources.length; ++ii) {
+                if (prevChunkSources != null) {
+                    prevChunkSources[ii].clear();
+                }
+            }
+
+            if (bufferChunkSources == null) {
+                // null out our current values
+                for (int ii = 0; ii < switchSources.length; ++ii) {
+                    //noinspection unchecked
+                    switchSources[ii].setNewCurrent((ColumnSource) nullColumnSources[ii]);
+                }
+            } else {
+                for (int ii = 0; ii < switchSources.length; ++ii) {
+                    switchSources[ii].setNewCurrent((ColumnSource) bufferChunkSources[ii]);
+                }
+            }
+            // current becomes prev, buffer becomes current, buffer is set to cleared prev for the next pass
```

Comment:
I find these almost exact textual descriptions of what the code right after does (which already use meaningful variable names to the point of literally sounding like reading the code) risky from a "code evolution" perspective while adding little value.  As in, you change the code below later, you are also forced to rewrite the comment.

My vote would be for a much more general "set the stage for next pass" comment (which is totally useful to separate the sections of what the code does, so I recognize the value of that in your comment).  But this is an opinion for your consideration, not a request for a change, I don't mind if you leave it.